controllers/agents/app/servicebinding_controller.go

```diff
@@ -501,11 +528,27 @@ func (r *ServiceBindingReconciler) removeVolumeMountFromContainer(ctx context.Co
 			}
 		}
 		for i, env := range c.Env {
+			var removeEnv bool
+			for _, bindingEnv := range sb.Spec.Env {
+				if env.Name == bindingEnv.Name {
+					if i < (len(c.Env) - 1) {
+						removeEnv = true
+					} else {
+						c.Env = c.Env[:i]
+					}
+				}
+			}
```

Comment:
what if an environment variable is removed from the ServiceBinding?
I think that if we just remove all the EnvVars that are referencing the ServiceEndpointDefinition Secret before L443 would work better. WDYT?
The ones defined in the new generation of the ServiceBinding will be added again at L443.

this should also supersede the previous comment

Reply:
may we also add this scenario in tests in this PR?

Reply:
If we remove the environment variable, everytime before the process starts, we might result in failing pods even when it is not required right.
And we might have an application which has environment variables other than what is projected by service binding.

Reply:
Finally, the idea is just to remove the variables from the Service Binding's Secret, not all the variables

Reply:
No, i mean the env variables not decalred by service binding specification

Reply:
We should just remove variables that are pointing to the service Binding's secret. The other ones (user defined) should be left there

Reply:
`I think that if we just remove all the EnvVars that are referencing the ServiceEndpointDefinition Secret before L443 would work better. WDYT?`... OK so you were suggesting to just remove all the env variables which match with the service binding spec, I interpreted that you menat to equate container env to nil